src/context/modalContext/ModalContext.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import {
+  ModalProvider,
+  useModal as rmhUseModal,
+} from 'react-modal-hook';
+
+export const ModalHookProvider = ({children}) => (
+  <ModalProvider>{children}</ModalProvider>
+);
+
+/**
+ * @param {JSX.Element} modal
+ * @param {any[]} [deps]
+ */
+export function useModal(modal, deps = null) {
+  const [openModal, closeModal] = rmhUseModal(() => {
+    const {type: ModalComponent, props, key} = modal;
+    return (
+      <ModalComponent key={key} {...props} closeModal={closeModal} />
```

Comment:
This is pretty nuanced, but I like to use the prop name `onClose` instead of `closeModal` since the provided method should be considered a side-effect of closing the modal, not the close event itself.  Namely because modals have several ways to be closed such as clicking a `ClickAwayListener`, clicking the `close` button, pressing the `esc` key, and clicking one or more controls inside of the modal body.  

Another great reason for this approach is that the modal hook should have ownership over what opening and closing the modal looks like.  If, for example, you wanted to change the URL to indicate that a particular modal was open, and then remove this URL on close, you would do that here in the custom `useModal` hook, not down inside of each implementation of the `closeModal` prop.

Concretely, this would look something similar to:

```
export function useModal(modal, deps = null) {
  const [openModal, closeModal] = rmhUseModal(() => {
    const {type: ModalComponent, props, key} = modal;
    const closeHandler = (onCloseFn) => {
      // So much more room for activities!
      closeModal()
      onCloseFn && onCloseFn()
    }
    const openHandler = (onOpenFn) => {
      // So much more room for activities!
      openModal()
      onOpenFn && onOpenFn()
    }
    return (
      <ModalComponent key={key} {...props} onClose={closeHandler} />
    );
  }, deps);
 return [openHandler, closeHandler];
} 
```

Reply:
I like it!
One question. Why `openHandler` and `closeHandler` can get callbacks? If someone wants to run a function after the modal is opened he can do it right after **he** calls to `openHandler`. For example, instead of:

```js
const [openModal] = useModal(<MyComponent />);
openModal(modalOpenedCallback);
```
do
```js
const [openModal] = useModal(<MyComponent />);
openModal();
modalOpenedCallback();
```

What am I missing?

Reply:
That way works too; I don't think you're missing anything.  In fact, you could allow callbacks and _still_ do what you have written here.  The callback approach is aimed at preventing an issue where someone didn't realize that the order mattered and did this by mistake:

``` javascript
const [openModal] = useModal(<MyComponent />);
modalOpenedCallback();
openModal();
```

When done as a callback you're ensuring that the code is called at the correct time by letting the parent function decide when that is.

The more important takeaway here is that handing over direct access to the `closeModal` and `openModal` functions provided by `react-modal-hooks` takes away your opportunity to add centralized `onOpen` and `onClose` side-effects to your modal.  

This could all fall under the umbrella of `YAGNI`, but I've seen so many people bitten by handing over direct access to a 3rd-party library call that it seems worth calling out when I see it.

Reply:
I like the thinking of own your own API rather than expose directly a 3rd-party's!

Actually, I can't return `[openHandler, closeHandler];` because `openHandler` and `closeHandler` are declared inside `rmhUseModal`'s callback so they are not available out of its scope.

Reply:
That's what I get for programming inside of a GitHub comment 🤦‍♂ 

Will this work?

``` javascript
export function useModal(modal, deps = null) {
  let closeHandler, openHandler;
  const [openModal, closeModal] = rmhUseModal(() => {
    const {type: ModalComponent, props, key} = modal;
    return (
      <ModalComponent key={key} {...props} onClose={closeHandler} />
    );
  }, deps);

    closeHandler = (onCloseFn) => {
      closeModal()
      onCloseFn && onCloseFn()
    }
    openHandler = (onOpenFn) => {
      openModal()
      onOpenFn && onOpenFn()
    }


 return [openHandler, closeHandler];
} 
```

No time to validate this unfortunately, but it's worth a shot if you want to get this working.  If this approach doesn't work, you can always duplicate the close handler or use some currying or something like that.

Reply:
😀Might be.
I already merged it so if you have some time I'll be happy if you could create a PR.
Thanks!